custom-commands/drupal-refresh/host/refresh

```diff
@@ -0,0 +1,176 @@
+    echo "> yes"
+  else
+    read -p "> " -r
+    printf '\n'
+    case $(echo "$REPLY" | tr '[A-Z]' '[a-z]') in
+      # On "y", "yes", or no input (default action), okay to proceed (no exit).
+      yes | y | "" ) ;;
+      # Any other response...
+      * )
+        if [ -n "$help_text" ] ; then
+          echo "$help_text"
+        fi
+        exit 0
+        ;;
+    esac
+  fi
+}
+
+# Use the first "positional parameter" as git branch. Note: All the options
+# have been popped from the args list, so only positional parameters are left.
+branch="$1"
+
+# Ensure no other positional parameters were passed.
+if [ "$#" -gt 1 ] ; then
+  usage_error "This command only takes one argument (git branch name), got \"$*\""
+fi
+
+# Check for invalid options combinations.
+if can_skip_download && (! can_import_db) ; then
+  usage_error 'The "--existing-sql" option requires "--import-db".'
+fi
+
+# Make sure an existing database file is present, if requested.
+db_dump_exists () { [ -f "${DDEV_APPROOT}/files/private/db.sql" ] ; }
+if can_import_db && can_skip_download && (! db_dump_exists) ; then
+  usage_error \
+    'There is no previously-downloaded database dump. Rerun this command without "--existing-sql" to download a new file.'
+fi
+
+# Confirm reverting Drupal configuration changes.
+has_config_changes () { ! ddev task drupal:config:check >/dev/null 2>&1 ; }
+if can_update_drupal && (! can_import_db) && has_config_changes ; then
+  printf '\nYour local environment has some Drupal configuration overrides:\n\n'
+  if ! ddev drush config:status ; then
+    usage_error 'Mismatch between code and database. Add the "--import-db" option to reinitialize the database.'
+  fi
+  confirm \
+    'Discard these configuration overrides and proceed?' \
+    'To preserve configuration changes, export them to code with "drush config:export", and either commit them to a different/temporary branch, or stash the changes with "git stash".'
+fi
+
+# Confirm importing the database.
+if can_import_db ; then
+  confirm \
+    'Importing the database replaces all content in your Drupal DB. Okay to discard local content changes?' \
+    'To skip the database import step, rerun this command without the "--import-db" option.'
+fi
+
+# Helper function to print command before executing it.
+announce () {
+  step_is_allowed="$1"
+  step="${@:2}"
+  if $step_is_allowed ; then
+    # Print the command first.
+    printf "\n\$ $step\n"
+    # Perform the step (execute the command).
+    $step
+    printf '\n'
+  elif [ $verbose ] ; then
+    echo "[info] Skipping \"$step\""
+  fi
+}
+
+# Abort the rest of the script if any command fails from here onward.
+set -e
+
+# Steps to refresh.
+announce can_change_branch   git fetch $verbose
+announce can_change_branch   git checkout "$branch" $verbose
+announce can_update_git      git pull $verbose
+announce can_restart_ddev    ddev restart
+announce can_build_composer  ddev task drupal:composer:development $verbose
```

Comment:
These `ddev task` commands rely on [Drainpipe](https://github.com/Lullabot/drainpipe), but can be removed or replaced with your own steps. E.g., in this case:
```sh
ddev composer install --optimize-autoloader
```